src/test_fixture/world.rs

```diff
@@ -245,72 +252,35 @@ where
         <[_; 3]>::try_from(output).unwrap()
     }
 
-    async fn malicious<'a, O, M, H, R, P>(&'a self, input: I, mut helper_fn: H) -> [O; 3]
+    async fn malicious<'a, O, M, H, R, P>(&'a self, input: I, helper_fn: H) -> [O; 3]
     where
         A: IntoMalicious<F, M>,
         O: Send + Debug,
         M: Send,
-        H: FnMut(MaliciousContext<'a, F>, M) -> R + Send,
+        H: Fn(MaliciousContext<'a, F>, M) -> R + Send + Sync,
```

Comment:
I may be wrong, but I think @martinthomson tried hard to keep it `Send` only

Reply:
I couldn't come up with a use case where it would be a problem for this to be Sync. In most cases the closure doesn't have any state at all, i.e., `helper_fn` could have type `fn(MaliciousContext<'a, F>, M) -> R`. There are a few tests in reshare.rs that do `for &role in Role::all()` and capture `role` in the closure. That case, or anything where the closure captures some static data, is still Sync.

Reply:
Yeah, I'm happy to see this code cleaned up like this.  The extra constraints don't seem to affect the code we have, so that's pretty strong evidence that we don't need the extra flexibility that you get from `FnMut` or non-`Sync` functions.  Same goes for `P` being `!Clone`.